bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -58,51 +61,57 @@ def optimizedparam(sampleval, nfold=2):
-mchirp_n = convert.mchirp_from_mass1_mass2(mass1_n, mass2_n)
-eta_n = convert.eta_from_mass1_mass2(mass1_n, mass2_n)
+#calling template data and constructing template_kde_hdf_file over same template params
+temp_file = h5py.File(args.template_file, 'r')
+approximant = temp_file['approximant'][:]
+f_lower = temp_file['f_lower'][:]
+tid = numpy.arange(len(f_lower))
+mass1, mass2, spin1z, spin2z = triggers.get_mass_spin(temp_file, tid)
+chi_eff = convert.chi_eff(mass1, mass2, spin1z, spin2z)
+mchirp = convert.mchirp_from_mass1_mass2(mass1, mass2)
+eta = convert.eta_from_mass1_mass2(mass1, mass2)
 
 assert len(args.fit_param) == len(args.log_param)
+parvals = []
 
-if args.make_template_kde:
-    param0 = eval(args.fit_param[0][:])
-    param1 = eval(args.fit_param[1][:])
-    param2 = eval(args.fit_param[2][:])
-    if args.log_param == ['False', 'False', 'False']:
-        sample_n = numpy.vstack((param0, param1, param2)).T
-    elif args.log_param == ['True', 'False', 'False']:
-        sample_n = numpy.vstack((numpy.log(param0), param1, param2)).T
-    elif args.log_param == ['True', 'True', 'False']:
-        sample_n = numpy.vstack((numpy.log(param0), numpy.log(param1), param2)).T
+for param, slog in zip(args.fit_param, args.log_param):
+    data = triggers.get_param(param, args, mass1, mass2, spin1z, spin2z)
+    if slog in ['false', 'False', 'FALSE']:
+        logging.info('Using param: %s', param)
+        parvals.append(data)
+    elif slog in ['true', 'True', 'TRUE']:
+        logging.info('Using log param: %s', param)
+        parvals.append(numpy.log(data))
     else:
-        raise ValueError("invalid log_param argument, use 'True', or 'False'")    
+        raise ValueError("invalid log param argument, use 'true', or 'false'")
+sample1 = numpy.vstack((parvals)).T
+grid_pts = sample1
 
-    grid_pts_n = sample_n
+if args.make_template_kde:
     logging.info('Starting optimization of template KDE parameters')
-    optbw_n, optalpha_n = optimizedparam(sample_n, nfold=2)
+    optbw, optalpha = optimizedparam(sample1, nfold=eval(args.nfold_template))
     logging.info('Evaluating template KDE')
-    kde_data_n = kde_awkde(sample_n, grid_pts_n, alp=optalpha_n, gl_bandwidth=optbw_n)    
+    kde_data = kde_awkde(sample1, grid_pts, alp=optalpha, gl_bandwidth=optbw)
 
     hdf = h5py.File(args.output_file, 'w')
-    hdf.create_dataset('kde_template', data=kde_data_n)
-    hdf.attrs['bank_kde'] = 'kde_template'
-    hdf.create_dataset('mass1', data=mass1_n)
-    hdf.create_dataset('mass2', data=mass2_n)
-    hdf.create_dataset('spin1z', data=spin1z_n)
-    hdf.create_dataset('spin2z', data=spin2z_n)
-    hdf.create_dataset('mchirp', data=mchirp_n)
-    hdf.create_dataset('eta', data=eta_n)
-    hdf.create_dataset('chi_eff', data=chi_eff_n)
-    hdf.create_dataset('grid_pts', data=grid_pts_n)
+    hdf.create_dataset('kde_temp', data=kde_data)
+    hdf.attrs['kde_tempplate'] = 'temp_file'
+    hdf.attrs['fit_param'] = 'args.fit_param'
```

Comment:
It looks like this creates an attribute with the name 'fit_param' and the value is the string 'args.fit_param'.  I don't see why recording this string is useful.  I think I suggested to make an attribute which records the fit params that the user has specified on the command line.  So that would require the _value_ of args.fit_param to be converted to a string.